backend/Tasque.BLL/Services/BoardService.cs

```diff
@@ -1,32 +1,117 @@
-﻿using AutoMapper;
-using FluentValidation;
-using Microsoft.EntityFrameworkCore;
+﻿using Microsoft.EntityFrameworkCore;
 using Tasque.Core.Common.DTO.Board;
-using Tasque.Core.Common.Entities;
+using Tasque.Core.Common.DTO.Task;
 using Tasque.Core.DAL;
+using Tasque.Core.Common.Entities;
+using System.Reflection.Metadata.Ecma335;
+using AutoMapper;
+using Tasque.Core.Common.DTO.User;
 
 namespace Tasque.Core.BLL.Services
 {
-    public class BoardService : EntityCrudService<Board>
+    public class BoardService
     {
-        private DataContext _context;
+        private DataContext _db;
         private IMapper _mapper;
 
-        public BoardService(DataContext context, IMapper mapper) : base(context)
+        public BoardService(DataContext context, IMapper mapper)
         {
-            _context = context;
+            _db = context;
             _mapper = mapper;
         }
 
-        public async Task<ICollection<BoardDto>> GetUserBoards(int userId)
+        public async Task<BoardInfoDto?> GetBoardByProjectId(int projectId)
         {
-            var userEntity = await _db.Users.FirstOrDefaultAsync(u => u.Id == userId)
-                  ?? throw new ValidationException("User not found");
+            var board = await _db.Boards
+                .FirstOrDefaultAsync(b => b.ProjectId == projectId);
+
+            if (board == null)
+            {
+                return null;
```

Comment:
Will it be clear to us what happens when null returns? Wouldn't it be better to throw an exception?

Reply:
fixed